hi-ml-cpath/src/health_cpath/utils/viz_utils.py

```diff
@@ -196,16 +204,34 @@ def plot_heatmap_overlay(
                                          location_bbox=slide_dict[SlideKey.ORIGIN],
                                          scale_factor=slide_dict[SlideKey.SCALE],
                                          should_upscale_coords=should_upscale_coords)
-    cmap = plt.cm.get_cmap("Reds")
+    cmap = plt.cm.get_cmap("Spectral_r")  # _r reverse the color map so that the highest attention is red
 
     tile_xs, tile_ys = sel_coords.T
     rects = [patches.Rectangle(xy, tile_size, tile_size) for xy in zip(tile_xs, tile_ys)]
 
-    pc = collection.PatchCollection(rects, match_original=True, cmap=cmap, alpha=0.5, edgecolor="black")
+    # line width is set to 0 to avoid the black border around the tiles as the tiles are already colored
+    pc = collection.PatchCollection(rects, match_original=True, cmap=cmap, alpha=0.5, linewidth=0)
     pc.set_array(np.array(attentions))
-    pc.set_clim([0, 1])
-    ax.add_collection(pc)
-    plt.colorbar(pc, ax=ax)
+    ax1.add_collection(pc)
+    plt.colorbar(pc, cax=cax)  # add colorbar to the right of the plot (cax)
+    return fig
+
+
+def plot_attention_histogram(case: str, slide_node: SlideNode, results: Dict[ResultsKey, List[Any]]) -> plt.Figure:
+    """Plots a histogram of the attention values of the tiles in a bag.
+
+    :param case: The report case (e.g., TP, FN, ...)
+    :param slide_node: The slide node that encapsulates the slide metadata.
+    :param results: Dict containing ResultsKey keys (e.g. slide id) and values as lists of output slides.
+    :return: matplotlib figure of the histogram of the attention values of the tiles in a bag.
+    """
+    slide_ids = [item[0] for item in results[ResultsKey.SLIDE_ID]]
+    slide_idx = slide_ids.index(slide_node.slide_id)
+    attentions = results[ResultsKey.BAG_ATTN][slide_idx]
+    fig, ax = plt.subplots()
+    ax.set_xlabel("Attention scores")
+    fig.suptitle(_get_histo_plot_title(case, slide_node))
+    ax.hist(attentions.cpu().numpy(), alpha=0.5)
```

Comment:
Fix the histogram x-range so that we can compare histograms more easily?

Reply:
OK, after reading your comments above it will probably be better to have a flexible range